enhancements/cluster-logging/elasticsearch-hot-warm-architecture.md

```diff
@@ -0,0 +1,156 @@
+approvers:
+  - "@jcantrill"
+  - "@bparees"
+  - "@alanconway"
+creation-date: 2020-10-13
+last-updated: 2020-10-13
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# elasticsearch-hot-warm-architecture
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Migration plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of this is to increase the potential retention of our logs and maintain that they can still be queried. Currently our logs are either maintained as a high priority index or they are deleted. We seek to emulate the back-end process that Elasticsearch makes available in the form of [x-pack API](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/index-lifecycle-management-api.html) end points.
+
+## Motivation
+
+Currently our managed Elasticsearch cluster is restricted in the amount of days of retention we can provide in part due to the amount of metadata ES keeps in memory related to its indices. If we are taking advantage of the different index management APIs provided by Elasticsearch we can seek to better tune clusters to increase retention and overall performance.
+
+### Goals
+The specific goals of this proposal are:
+
+* Outline the addition to the API to provide a means to configure WARM policies
+
+* Replace|repurpose the current rollover cronjob to also perform hot to warm transition actions
+
+We will be successful when:
+
+* Users are able to configure WARM policies so that we can optimize their cluster usage
+
+* We are able to transition a user's index from one policy to the next based on their configuration
+
+### Non-Goals
+
+* We will not be marking different nodes for the different policies and reallocating shards based on that
+
+* We will not be implementing this using the Opendistro plugin
+
+## Proposal
+
+Currently we offer a mechanism for users to configure their index lifecycles based on two states: hot and delete. By default indices are created and stay in the hot state, where they can be rolled over based on a configuration, however they stay in a "hot" state until they are eventually deleted based on the delete policy.
+
+Elasticsearch offers mechanisms to move indices between hot, warm, cold, and delete and discusses approaches for this [in a blog article](https://www.elastic.co/blog/implementing-hot-warm-cold-in-elasticsearch-with-index-lifecycle-management). However, this functionality is a x-pack feature and we do not support these. [Opendistro](https://opendistro.github.io/for-elasticsearch-docs/docs/ism/) offers a way to do this as well, however their plugins require ES7. We achieve our current functionality using cronjobs -- one for delete and one for rollover.
```

Comment:
The penultimate sentence is not clear enough to me. AFAICS the link provided for ISM speaks about Kibana or is marked as Kibana. Does our OpenDistro ES6 miss the entire ISM API or do we miss here the plugin for Kibana only?

Reply:
There is a means to configure it via Kibana, however it is not required for ISM. When I talk about ISM in the context of this document is it with respect to ES as that is how we would be configuring it for customers.